### PR TITLE
Improve vmware_ops integration tests cleanup implementation

### PR DESCRIPTION
**Pull request by anna-savina**
Jira ticket - [ACA-1695](https://issues.redhat.com/browse/ACA-1695)

Changes included (separated by commits):

- Added `retries` to eliminate random failures
- Fixed `always` block for `vmware_ops_manage_folder_test`
- Added `ignore_errors` to `cleanup_esxi.yml` of `vmware_ops_provision_virtual_esxi_test`
- Added `tiny_prefix` to isolate test runs
- Fixed assertion in `vmware_ops_provision_vm_test`

### CODE REVIEW

**Review by shellymiron — APPROVED** [review]

File: tests/integration/targets/vmware_ops_deploy_ovf_test/tasks/main.yml [hunk 1]
```diff
@@ -82,11 +82,19 @@
...
+      until: result.failed == false
 
     - name: Delete OVF On Local Filesystem
       ansible.builtin.file:
         state: absent
         path: "{{ ovf_template_local | dirname }}"
+      retries: 5
+      delay: 5
+      register: result
+      until: result.failed == false
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_deploy_ovf_test/tasks/main.yml:97]: I think it's not needed here, it's just removing a file on local filesystem

File: tests/integration/targets/vmware_ops_export_vm_as_ovf_test/tasks/main.yml [hunk 2]
```diff
@@ -47,8 +47,16 @@
...
+      until: result.failed == false
 
     - name: Delete Exported OVF
       ansible.builtin.file:
         state: absent
         path: "{{ export_vm_as_ovf_export_dir }}"
+      retries: 5
+      delay: 5
+      register: result
+      until: result.failed == false
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_export_vm_as_ovf_test/tasks/main.yml:62]: same here as it is a local file

File: tests/integration/targets/vmware_ops_provision_virtual_esxi_test/vars/main.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ provision_virtual_esxi_datacenter: "{{ vcenter_datacenter }}"
 provision_virtual_esxi_datastore_iso_path: "{{ custom_esxi_8_iso_path }}"
 resource_pool_name: ci-resource-pool-test
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_provision_virtual_esxi_test/vars/main.yml:13]: I guess `tiny_prefix` should be added to `resource_pool_name` as well

File: tests/integration/targets/vmware_ops_provision_vm_test/vars/main.yml [hunk 4]
```diff
@@ -71,7 +71,7 @@ provision_vms:
     provision_vm_guest_id: "rhel9_64Guest"
     provision_vm_datastore: "datastore1"
   # RHEL8 VM
-  - provision_vm_name: "qe-provision-vm-rhel-8"
+  - provision_vm_name: "{{ tiny_prefix }}-qe-provision-vm-rhel-8"
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_provision_vm_test/vars/main.yml:74]: please add `tiny_prefix` as well to

-  `portgroup_name` [here](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/tests/integration/targets/vmware_ops_provision_vm_test/vars/main.yml#L123) 
-  `vm_updated_name` [here](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/tests/integration/targets/vmware_ops_provision_vm_test/vars/main.yml#L104)

File: tests/integration/targets/vmware_ops_esxi_maintenance_mode_test/vars/main.yml [hunk 5]
```diff
@@ -16,7 +16,7 @@ provision_virtual_esxi_folder: ""
 provision_virtual_esxi_resource_pool: "{{ resource_pool_name }}"
 provision_virtual_esxi_datastore_iso_path: "{{ custom_esxi_8_iso_path }}"
 provision_virtual_esxi_vms:
-  - name: ci-vcenter-maintenance-test
+  - name: "{{ tiny_prefix }}-vm-esxi_maintenance-test"
 provision_virtual_esxi_networks:
     - name: "{{ test_network_name }}"
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_esxi_maintenance_mode_test/vars/main.yml:21]: I guess we should add `tiny_prefix` to `test_network_name` in group_vars.yml at https://github.com/redhat-cop/cloud.vmware_ops/blob/main/tests/integration/targets/group_vars.yml#L13

File: tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml [hunk 6]
```diff
@@ -1,64 +1,65 @@
...
-    manage_folder_state: present
-    manage_folder_folder_name: production/foo/web
-    manage_folder_folder_type: "{{ item }}"
+    - name: Create Folder Tree
+      loop: "{{ folder_types }}"
+      ansible.builtin.include_role:
+        name: cloud.vmware_ops.manage_folder
+      vars:
+        manage_folder_state: present
+        manage_folder_folder_name: production/foo/web
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml:15]: please add `tiny_prefix`

File: tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml [hunk 7]
```diff
@@ -1,64 +1,65 @@
...
-    manage_folder_parent_folder: production/foo
-    manage_folder_folder_type: "{{ item }}"
+    - name: Create Folder Without Managing Full Tree
+      loop: "{{ folder_types }}"
+      ansible.builtin.include_role:
+        name: cloud.vmware_ops.manage_folder
+      vars:
+        manage_folder_state: present
+        manage_folder_folder_name: db
+        manage_folder_parent_folder: production/foo
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml:25]: please add `tiny_prefix`

File: tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml [hunk 8]
```diff
@@ -1,64 +1,65 @@
...
-    manage_folder_parse_name_as_path: false
-    manage_folder_folder_type: "{{ item }}"
+    - name: Create A Folder With A Slash In It
+      loop: "{{ folder_types }}"
+      ansible.builtin.include_role:
+        name: cloud.vmware_ops.manage_folder
+      vars:
+        manage_folder_state: present
+        manage_folder_folder_name: security/syseng
+        manage_folder_parent_folder: production/foo
```
Comment (elsapassaro) [inline tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml:35]: please add `tiny_prefix`, maybe better to create a new variable in vars/main.yml for it.
add `tiny_prefix` also to cleanup step [here](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml#L68)

File: tests/integration/targets/vmware_ops_deploy_ovf_test/tasks/main.yml (hunk 1 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_deploy_ovf_test/tasks/main.yml]: done

File: tests/integration/targets/vmware_ops_export_vm_as_ovf_test/tasks/main.yml (hunk 2 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_export_vm_as_ovf_test/tasks/main.yml]: done

File: tests/integration/targets/vmware_ops_provision_virtual_esxi_test/vars/main.yml (hunk 3 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_provision_virtual_esxi_test/vars/main.yml]: done

File: tests/integration/targets/vmware_ops_provision_vm_test/vars/main.yml (hunk 4 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_provision_vm_test/vars/main.yml:74]: done

File: tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml (hunk 6 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml]: done

File: tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml (hunk 7 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml]: done

File: tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml (hunk 8 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_manage_folder_test/tasks/main.yml]: done

File: tests/integration/targets/vmware_ops_esxi_maintenance_mode_test/vars/main.yml (hunk 5 above)
Comment (anna-savina) [reply tests/integration/targets/vmware_ops_esxi_maintenance_mode_test/vars/main.yml:21]: since `TestNetwork` is permanent, no need to add `tiny_prefix`

**Review by elsapassaro — APPROVED** [review]

**Pull request merged**